airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableRow.tsx

```diff
@@ -46,38 +49,74 @@ export const CatalogTreeTableRow: React.FC<StreamHeaderProps> = ({
     [syncMode, destinationSyncMode]
   );
 
+  const { initialValues } = useConnectionFormService();
```

Comment:
No more prop drilling to find out if the row changed!  We can do this at the row level now that we have our context.

This will also make it relatively straightforward to grab the diff from the context and apply it to some of the conditional logic below when we get to the point where we want to integrate the diff on here 💥